src/engine/sparqlExpressions/RelationalExpressions.cpp

```diff
@@ -434,21 +434,44 @@ RelationalExpression<comp>::getPrefilterExpressionForMetadata(
   const SparqlExpression* child0 = children_.at(0).get();
   const SparqlExpression* child1 = children_.at(1).get();
 
-  const auto tryGetPrefilterExprVariablePairVec =
-      [](const SparqlExpression* child0, const SparqlExpression* child1,
-         bool reversed) -> std::vector<PrefilterExprVariablePair> {
-    const auto* variableExpr = dynamic_cast<const VariableExpression*>(child0);
-    if (!variableExpr) {
-      return {};
+  const auto getOptVariableAndIsYear = [](const SparqlExpression* child)
```

Comment:
Add a small comment about what this does. 
And this lambda doesn't capture, so it can be a free function (less cluttering in the code).